pkg/dbal/DbalProducer.php

```diff
@@ -125,10 +125,14 @@ public function send(Destination $destination, Message $message): void
                 'queue' => Type::STRING,
                 'time_to_live' => Type::INTEGER,
                 'delayed_until' => Type::INTEGER,
-                'redelivered' => Type::BOOLEAN,
+                'redelivered' => Type::SMALLINT,
```

Comment:
This is strange. Functional tests are working with the current code, and as far as I know doctrine should handle boolean type internally according to platform driver.

Reply:
IMO, should be investigated deeper.

Reply:
I found people had similar problems with boolean. For eample https://stackoverflow.com/questions/10242312/pdo-bindvalue-with-pdoparam-bool-causes-statement-execute-to-fail-silently

Try to search "insert with PDO::PARAM_BOOL fails"
I have php 7.3 and mysql 5.7, but the issue is still here.

May be if you could use SMALLINT everywhere for the type of this field it could be a solution.